nrf52-hal-common/src/spim.rs

```diff
@@ -29,8 +30,71 @@ use embedded_hal::digital::v2::OutputPin;
 /// - The SPIM instances share the same address space with instances of SPIS,
 ///   SPI, TWIM, TWIS, and TWI. You need to make sure that conflicting instances
 ///   are disabled before using `Spim`. See product specification, section 15.2.
+#[derive(Debug)]
 pub struct Spim<T>(T);
```

Comment:
In which cases does `T` implement `Debug` here?

Reply:
I use this pattern frequently in my own code bases when writing tests.  The `T` would be a mocked instance that debug prints all of the current pin states to aid in debugging, for example.  I've also developed muscle memory to always derive `Debug` because it is quite common to `#[deny(missing_debug_implementations)]` :)

Reply:
Ah, that sounds really useful actually!